robot_skills/src/robot_skills/amigo.py

```diff
@@ -1,9 +1,33 @@
 #! /usr/bin/env python
 
 import robot
+import rospy
+import tf
+import math
+
+from geometry_msgs.msg import PoseStamped
+
+
+class SSL(object):
+    def __init__(self, topic):
+        self._sub = rospy.Subscriber(topic, PoseStamped, self._callback, queue_size=1)
+        self._last_msg = None
+        self._last_received_time = rospy.Time(0)
+
+    def _callback(self, msg):
+        self._last_msg = msg
+        self._last_received_time = rospy.Time.now()
+
+    def get_last_yaw(self, max_age_seconds=2):
```

Comment:
This API is really specific to the challenge. You can better expose the callback (with a yaw for example)